server/utils/addOverbookingsToSchedule.ts

```diff
@@ -0,0 +1,41 @@
+import { BedOccupancyEntryUi } from '@approved-premises/ui'
+import { addDays, differenceInDays, subDays } from 'date-fns'
+
+export const addOverbookingsToSchedule = (schedule: Array<BedOccupancyEntryUi>): Array<BedOccupancyEntryUi> => {
+  const overbookings: Array<BedOccupancyEntryUi> = []
+
+  // Ensure the schedule is ordered by date
+  const orderedSchedule = schedule.sort((a, b) => a.startDate.getTime() - b.startDate.getTime())
+
+  // Loop through each schedule item and check for overbookings
+  const newSchedule = orderedSchedule.map((item, index) => {
+    const nextItem = orderedSchedule[index + 1]
+    // Is the next item a lost bed or booking?
+    if (nextItem && ['booking', 'lost_bed'].includes(nextItem.type)) {
+      // If it is, is the start date before our end date?
+      if (nextItem.startDate < item.endDate) {
```

Comment:
Is there a reason we couldn't use `isBefore` here?

Reply:
Absolutely no reason! Only reason was because the code was making my eyes bleed! 😆